File: webflux/src/main/java/ru/com/m74/example/webflux/controller/DefaultController.java
package ru.com.m74.example.webflux.controller;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;
import ru.com.m74.example.webflux.dto.Example;

@RestController
public class DefaultController {

    @GetMapping("/")
    public Mono<String> index() {
        return Mono.just("Hello world");
    }

    @GetMapping("/flux")
    public Flux<Example> flux() {
        return Flux.just("one", "two").map(Example::new);
    }
}

File: kotlin/src/main/kotlin/ru/com/m74/example/kotlin/controller/DefaultController.kt
package ru.com.m74.example.kotlin.controller

import org.slf4j.LoggerFactory
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import java.time.LocalDateTime

@RestController
@RequestMapping("/")
class DefaultController() {

    val log = LoggerFactory.getLogger(javaClass)

    interface Message {
        fun getExampe() = "Example"
    }

    @GetMapping
    fun index() = object : Message {
        val text
            get() = "Hello world!!!"
        val now = LocalDateTime.now()

        init {
            log.info("init message: {}", this)
        }
    }

}